src/database.c

```diff
@@ -72,6 +74,7 @@ static pthread_mutex_t update_lock;
 #define READ_TEST_KEY	1
 #define MEGABYTE	1024*1024
 #define DATA_FORMAT "%i %lu %s"
+#define MAX_PATH_LEN 490 /* MDB_MAXKEYSIZE = 511 bytes */
```

Comment:
What's the point of defining MAX_PATH_LEN? Why not just checks against MDB_MAXKEYSIZE in your conditional statement? It makes it more transparent as to why you're taking a particular path in that case. Unless there's something hidden here that I'm not aware of?

Reply:
Unfortunately, according to the LMDB documentation, MDB_MAXKEYSIZE is defined in a source file(.c), so it's not available here. I did some digging and figured out that there is a function called ```int mdb_env_get_maxkeysize(MDB_env* env)``` that should do the trick.